### PR TITLE
tests: use --headless=new for all smoketests

### PR DESCRIPTION
**Pull request by paulirish**
I've wanted to use headless for CLI smoketests to avoid windows popping up and stealing focus. But as we know, some test assertions fail.

The `--headless=chrome` flag provides something much more like Chrome than what the `--headless` flag launches. And it now has support across windows/linux/mac. 

edit: feb 2023: the flag was renamed to --headless-new. https://github.com/GoogleChrome/chrome-launcher/pull/290

🔒 http://go/rooxt and http://go/wkigd

### CODE REVIEW

File: cli/test/smokehouse/lighthouse-runners/cli.js [hunk 1]
```diff
@@ -61,6 +61,8 @@ async function internalRun(url, tmpPath, configJson, options) {
     `${url}`,
     `--output-path=${outputPath}`,
     '--output=json',
+    // native headless, as full headless has several behavior differences that break tests. http://go/rooxt
+    '--chrome-flags="--headless=chrome"',
```
Comment (connorjclark) [inline cli/test/smokehouse/lighthouse-runners/cli.js:65]: Looking at my (stale) branch doing the same, here are some things to consider:

- there is also `lighthouse-runners/bundle.js` to update
- headless-chrome.md could mention this new mode. what I had:

> +There is also the new `--headless=chrome` option, which includes functionality that
> +was explicitly omitted from the original headless browser.
> +
>  ## CLI (xvfb)

**Review by adamraine — COMMENTED** [review]
Could we add a way to force headfull mode via the smokehouse CLI? `--force-headfull` perhaps?

File: cli/test/smokehouse/lighthouse-runners/cli.js (hunk 1 above)
Comment (paulirish) [reply cli/test/smokehouse/lighthouse-runners/cli.js:65]: SG. also i wanted to experiment with using this in CI and getting rid of xvfb, which should (in theory) work.

but i'll do this part in a followup as i expect it'll be slightly more involved

ill address your two bullets in this one

**Comment by paulirish** [comment]
> Could we add a way to force headfull mode via the smokehouse CLI? `--force-headfull` perhaps?

done.

-------------

also. i suspect theres still failures here. I'm gonna pick this PR back up post 10.0 probably.

File: types/internal/smokehouse.d.ts [hunk 2]
```diff
@@ -56,14 +56,19 @@ declare global {
...
 
     export interface SmokehouseOptions {
-      /** If true, performs extra logging from the test runs. */
-      isDebug?: boolean;
+      /** Options to pass to the specific Lighthouse runner. */
+      testRunnerOptions?: {
+        /** If true, performs extra logging from the test runs. */
+        isDebug?: boolean;
+        /** Launch Chrome in typical desktop headful mode, rather than our default of `--headless=new`. */
+        forceHeadful?: boolean;
```
[the remark this answers is not in the document]
Comment (adamraine) [reply types/internal/smokehouse.d.ts:64]: Would that still be `--force-headful` on the CLI though? If that's the way we are naming the flag I would prefer to just keep the option as `forceHeadful` throughout.

**Pull request merged**